### PR TITLE
feat: [SRM-11178]: fixing all scenarios related to unsaved changes for Verify step

### PR DESCRIPTION
**Pull request by harshilbits**
### Summary
1. Fixing all scenarios related to unsaved changes for Verify step

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by harshilbits** [comment]
retrigger coverage

**Review by praneshg239 — APPROVED** [review]

File: src/modules/85-cv/components/PipelineSteps/ContinousVerification/components/ContinousVerificationWidget/components/ContinousVerificationWidgetSections/components/ConfiguredMonitoredService/ConfiguredMonitoredService.utils.ts [hunk 1]
```diff
@@ -6,8 +6,17 @@
...
 
 export function isMonitoredServiceFixedInput(monitoredServiceRef: string): boolean {
   return !!(monitoredServiceRef !== RUNTIME_INPUT_VALUE && monitoredServiceRef && !isAnExpression(monitoredServiceRef))
 }
+
+export function isFirstTimeOpenForConfiguredMonitoredSvc(
+  formValues: ContinousVerificationData,
+  monitoredServiceData: ResponseMonitoredServiceResponse | null
+): boolean {
+  return !!(!formValues?.spec?.monitoredServiceRef && monitoredServiceData?.data?.monitoredService?.identifier)
```
[the remark this answers is not in the document]
Comment (praneshg239) [reply src/modules/85-cv/components/PipelineSteps/ContinousVerification/components/ContinousVerificationWidget/components/ContinousVerificationWidgetSections/components/ConfiguredMonitoredService/ConfiguredMonitoredService.utils.ts:21]: @deepesh-ui FYI

[the remark this answers is not in the document]
Comment (harshilbits) [reply src/modules/85-cv/components/PipelineSteps/ContinousVerification/components/ContinousVerificationWidget/components/ContinousVerificationWidgetSections/components/ConfiguredMonitoredService/ConfiguredMonitoredService.utils.ts:21]: Good point @praneshg239 . I will also check this in detail , just thinking won't `Boolean` internally do the same thing as double bang operator.

After checking this in detail , may we can adopt this practice.

**Review by vasuki-kalluri — APPROVED** [review]

**Review by amitsh14 — APPROVED** [review]

File: src/modules/85-cv/components/PipelineSteps/ContinousVerification/components/ContinousVerificationWidget/ContinousVerificationWidget.utils.ts [hunk 2]
```diff
@@ -249,3 +250,20 @@ export function validateMetricDefinitions(
 export function shouldValidateTemplateInputs(templateInputsToValidate: any, key: string, templateInputs: any): boolean {
   return templateInputsToValidate[key] !== 'object' && !templateInputs[key]
 }
+
+export function resetFormik(
+  formValues: ContinousVerificationData,
+  newSpecs: {
+    monitoredServiceRef?: string | undefined
```
Comment (reetika04) [inline src/modules/85-cv/components/PipelineSteps/ContinousVerification/components/ContinousVerificationWidget/ContinousVerificationWidget.utils.ts:257]:
```suggestion
    monitoredServiceRef?: string ```

**Pull request merged**